scapy/pton_ntop.py

```diff
@@ -75,15 +83,7 @@ def inet_ntop(af, addr):
                 hexstr = hex(value)[2:]
             except TypeError:
                 raise Exception("Illegal syntax for IP address")
-            parts.append(hexstr.lstrip("0").lower())
-        result = ":".join(parts)
-        while ":::" in result:
-            result = result.replace(":::", "::")
-        # Leaving out leading and trailing zeros is only allowed with ::
-        if result.endswith(":") and not result.endswith("::"):
-            result = result + "0"
-        if result.startswith(":") and not result.startswith("::"):
-            result = "0" + result
-        return result
-    else:
-        raise Exception("Address family not supported yet")        
+            parts.append(hexstr.lower())
+
+        # Note: the returned address is never compact
+        return ":".join(parts)
```

Comment:
Thanks for your proposal. As converting packet IPv6 to strings is full of corner cases, it is better to use socket.ntop and socket.ntop when possible.

If the current patch does not break tests on Windows, I will rather fall back to uncompressed IPv6 addresses at the last resort. I think that this will only happen when there is no IPv6 support ons the operating system. The only use case for these Scapy specific functions will be to send/receive IPv6 packets on an OS that does not support IPv6. I believe that this users can leave with uncompressed IPv6 addresses.